### PR TITLE
Update with review and approval from Fred.

### PR DESCRIPTION
**Pull request by jmframe**
[Short description explaining the high-level reason for the pull request]

## Additions

- Added "forcing_code" and test runs with passed Forcings/PET through BMI. 
- Review from Fred Ogden

## Removals

-

## Changes

-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/bmi_cfe.c [hunk 1]
```diff
@@ -746,17 +744,22 @@ static int Update (Bmi *self)
...
     
     cfe_ptr->vol_struct.volin += cfe_ptr->timestep_rainfall_input_m;
-
+    // Delete me...    printf("_______PRECIP IN  CFE UPDATE FUNCTION________: %lf\n", cfe_ptr->aorc.precip_kg_per_m2);
     run_cfe(cfe_ptr);
 
     // Advance the model time 
     cfe_ptr->current_time_step += 1;
 
-    // Set the current rainfall input to the right place in the forcing.
```
Comment (madMatchstick) [inline src/bmi_cfe.c:757]: update `get_value_ptr()` def to include `RAIN_RATE`

File: README.md [hunk 2]
```diff
@@ -11,11 +11,23 @@ There are actually multiple ways to run the CFE.
 The BMI functionality was developed as a standalone module in C. To compile this code the developer used these steps:
```
Comment (madMatchstick) [inline README.md:11]: typos (line 6): 1. As written by the ~~origional~~ original author. This ~~include~~ includes a full program...

File: README.md [hunk 3]
```diff
@@ -11,11 +11,23 @@ There are actually multiple ways to run the CFE.
...
+1. `module load gnu/10.1.0`
+2. `gcc -lm ./src/main_pass_forcings.c ./src/cfe.c ./src/bmi_cfe.c ./forcing_code/src/aorc.c ./forcing_code/src/bmi_aorc.c  -o run_cfe_bmi_pass_forcings`  
+This should generate an executable called **run_cfe_bmi_pass_forcings**. To run this executable you must pass the path to the corresponding configuration files for **BOTH** CFE and AORC (in that order): `./run_cfe_bmi_pass_forcings ./configs/cat_89_bmi_config_cfe_pass.txt ./configs/cat_89_bmi_config_aorc.txt`
+
+## CFE Model gets forcings AND Potential Evapotranspiration passed from BMI
+The CFE has functionality to remove mass through evapotransiration (directly from precipitation and from soil using the Bydukko function). The steps to test this functionality are very similar to the example above, but with just adding two additional src files, which come from the AORC BMI Module in the `forcing_code` directory.
```
Comment (madMatchstick) [inline README.md:26]: typo: evapotranspiration

File: README.md [hunk 4]
```diff
@@ -11,11 +11,23 @@ There are actually multiple ways to run the CFE.
...
+
+## CFE Model gets forcings AND Potential Evapotranspiration passed from BMI
+The CFE has functionality to remove mass through evapotransiration (directly from precipitation and from soil using the Bydukko function). The steps to test this functionality are very similar to the example above, but with just adding two additional src files, which come from the AORC BMI Module in the `forcing_code` directory.  
+1. `module load gnu/10.1.0`
+2. `gcc -lm ./src/main_cfe_aorc_pet.c ./forcing_code/src/pet.c ./forcing_code/src/bmi_pet.c ./src/cfe.c ./src/bmi_cfe.c ./forcing_code/src/aorc.c ./forcing_code/src/bmi_aorc.c -o run_cfe_aorc_et_bmi`  
+This should generate an executable called **run_cfe_bmi_pass_forcings**. To run this executable you must pass the path to the corresponding configuration files for **ALL** CFE, PET and AORC (in that order): `./run_cfe_aorc_et_bmi ./configs/cat_89_bmi_config_cfe_pass.txt ./configs/cat_89_bmi_config_aorc.txt ./configs/cat_89_bmi_config_pet_pass.txt`
+
 # Making this code to run in the NGen Framework
```
Comment (madMatchstick) [inline README.md:31]: suggestion?: Run CFE BMI in stand alone, independent of a framework

File: README.md [hunk 5]
```diff
@@ -11,11 +11,23 @@ There are actually multiple ways to run the CFE.
...
 make clean  
 make (the resulting executable is moved to parent directory in the Makefile script)
```
Comment (madMatchstick) [inline README.md:33]: typos later in doc: 
- ln 38: The CFE was based on the ~~t-short~~ t-shirt approximation...
- ln 39: t-shirt approximation of the hydrologic routing ~~funtionality~~ functionality of...
- ln 43: ~~Thesed~~ These are:
- ln 51: ...using a geomorphological ~~instantanteous~~ instantaneous unit hydrograph...
- ln 53: ...delayed response, and ~~elminates~~ eliminates the need...

File: README.md [hunk 6]
```diff
@@ -11,11 +11,23 @@ There are actually multiple ways to run the CFE.
 The BMI functionality was developed as a standalone module in C. To compile this code the developer used these steps:
 1. `module load gnu/10.1.0`
 2. `gcc -lm ./src/main.c ./src/cfe.c ./src/bmi_cfe.c -o run_bmi`
```
Comment (madMatchstick) [inline README.md:13]: change out exe to `run_cfe_bmi`

File: README.md (hunk 4 above)
Comment (madMatchstick) [inline README.md:31]: Actually - this is redundant to [read-local-forcing-file](https://github.com/jmframe/cfe#read-local-forcing-file).  You can keep `make` as build option but list under this section as an alternative to `./make_and_run_bmi.sh`.  Or just remove it all together...

**Review by madMatchstick — APPROVED** [review]
Approved - but see minor comments in README.  Build/run looks good for all three shell scripts. Also tested in ngen with no errors.  Great work!

File: README.md (hunk 4 above)
Comment (jmframe) [reply README.md:31]: I'm not sure I understand. Maybe make an issue for this one, if it is a redundancy.

**Pull request merged**